clang/include/clang/CIR/Interfaces/ASTAttrInterfaces.td

```diff
@@ -0,0 +1,213 @@
+//===- ASTAttrInterfaces.td - CIR AST Interface Definitions -----*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_CIR_INTERFACES_AST_ATTR_INTERFACES
+#define MLIR_CIR_INTERFACES_AST_ATTR_INTERFACES
+
+include "mlir/IR/OpBase.td"
+
+let cppNamespace = "::mlir::cir" in {
+
+    def ASTDeclInterface : AttrInterface< "ASTDeclInterface" > {
+        let methods = [
```

Comment:
IIUC, these methods expose Clang AST APIs as a service interface to CIR code. This is great as AST changes would  be less disruptive to CIR. I guess the list will need to grow as more AST APIs are used. BTW, how is a direct field access (without a getter/setter) supported?

Reply:
It is not you need to provide API in the interface, which exposes the field.